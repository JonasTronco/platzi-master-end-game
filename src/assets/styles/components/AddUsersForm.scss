@import '../main.scss';

.add-users {
  display: flex;

  &__img {
    width: 45px;
    height: 45px;
    margin-right: $space;
    fill: $secondary_color;
  }
}
.add-users-form {
  @include card;
  width: 100%;
  min-height: 170px;
  margin-top: $space_x6;
  display: flex;
  align-items: center;
  justify-content: flex-start;

  &__container {
    margin-left: $space_x6;
    max-width: 350px;
    height: 120px;
  }
  &__input {
    margin-bottom: $space_x6;
  }
  &__button {
    @include btn;
    background-color: $secondary_color;
    margin: 0 36%;
  }
}
