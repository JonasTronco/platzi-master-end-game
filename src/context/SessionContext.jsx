import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
