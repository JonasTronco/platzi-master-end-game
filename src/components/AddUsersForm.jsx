import { Formik, Form, Field, ErrorMessage } from 'formik';
import '../assets/styles/components/AddUsersForm.scss';
import UsersAddIcon from '../assets/icons/users-add.svg';

function AddUsersForm() {
  return (
    <>
      <div className='add-users'>
        <UsersAddIcon className='add-users__img' />
        <h1 className='add-users__title'>Add Users</h1>
      </div>
      <div className='add-users-form'>
        <Formik>
          <Form className='add-users-form__container lg'>
            <div className='add-users-form__content input gl'>
              <strong className='add-users-form__title  input__label'>
                Select a CSV file:
              </strong>
              <Field className=' add-users-form__input input__field lg' />

              <ErrorMessage
                name='upload'
                component='div'
                className='input__error'
              />
            </div>
            <button className='add-users-form__button' type='submit'>
              Upload
            </button>
          </Form>
        </Formik>
      </div>
    </>
  );
}
export default AddUsersForm;
