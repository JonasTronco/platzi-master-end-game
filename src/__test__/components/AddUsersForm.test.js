import AddUsersForm from '../../components/AddUsersForm';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('AddUsersForm Component', () => {
  test('Should render AddUsersForm component', () => {
    const component = create(
      <SessionProvider>
        <AddUsersForm />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
