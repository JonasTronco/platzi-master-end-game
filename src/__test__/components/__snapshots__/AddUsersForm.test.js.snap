// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUsersForm Component Should render AddUsersForm component 1`] = `
<form
  className="add-users-form"
>
  <h3
    className="add-users-form__title"
  >
    Upload a .csv file with users data
  </h3>
  <div
    className="input"
  >
    <label
      className="input__label"
    >
      Users Data File
      <input
        className="input__field--img"
        onChange={[Function]}
        type="file"
      />
    </label>
    
  </div>
  <button
    className="btn--disabled"
    onClick={[Function]}
    type="button"
  >
    Add Users
  </button>
</form>
`;
